manifests/application/migrator.pp

```diff
@@ -116,168 +123,326 @@
+        $primary_client_ca = $primary_client_ca_file_path
       }
-    }
 
-    if($primary_client_use_cert_auth and $primary_client_cert_source){
-      if(!$primary_client_cert_file_path){
-        fail('Must provide $primary_client_cert_file_path in order to deploy cert file.')
+      #ensure the primary client cert exists
+      if($primary_client_cert_source){
+        if($primary_client_cert_file_path){
+          $primary_client_cert = $primary_client_cert_file_path
+        }
+        else{
+          $primary_basename = basename($primary_client_cert_source)
+          $primary_client_cert = "$migrator_config_dir/$primary_basename"
+        }
+
+        file { $primary_client_cert:
+          mode    => '0644',
+          owner   => $service_owner,
+          group   => $service_group,
+          links   => 'follow',
+          source  => $primary_client_cert_source,
+          require => File[$migrator_config_dir],
+          notify  => [Service[$migrator_service_name]],
+        }
       }
+      else{
+        if(!$primary_client_cert_file_path){
+          fail('If using your own primary migrator cert then the $primary_client_cert_file_path must be provided.')
+        }
 
-      file { $primary_client_cert_file_path:
-        mode    => '0644',
-        owner   => $service_owner,
-        group   => $service_group,
-        links   => 'follow',
-        source  => $primary_client_cert_source,
-        notify  => [Service[$migrator_service_name]],
+        $primary_client_cert = $primary_client_cert_file_path
       }
     }
+    else {
+      $primary_client_ca = undef
+      $primary_client_cert = undef
+    }
 
+    $primary_config_file = "$migrator_config_dir/primary-lightblue-client.properties"
     lightblue::client::configure { $primary_config_file:
       owner                   => $service_owner,
       group                   => $service_group,
       lbclient_metadata_uri   => $primary_client_metadata_uri,
       lbclient_data_uri       => $primary_client_data_uri,
       lbclient_use_cert_auth  => $primary_client_use_cert_auth,
-      lbclient_ca_file_path   => $primary_client_ca_file_path,
-      lbclient_cert_file_path => $primary_client_cert_file_path,
+      lbclient_ca_file_path   => $primary_client_ca,
+      lbclient_cert_file_path => $primary_client_cert,
       lbclient_cert_password  => $primary_client_cert_password,
       lbclient_cert_alias     => $primary_client_cert_alias,
+      require                 => File[$migrator_config_dir],
       notify                  => [Service[$migrator_service_name]],
     }
 
-    if($source_config_file != $primary_config_file){
+    #configure source lightblue instance
+    if($is_source_same_as_primary){
+      $source_config_file = $primary_config_file
+    }
+    else {
       if !$source_client_metadata_uri or !$source_client_data_uri {
         fail('If defining a source_config, then you must also define data and metadata urls for it.')
```

Comment:
Ignore this, can tackle with https://github.com/lightblue-platform/lightblue-puppet/issues/125